pilot/pkg/model/context.go

```diff
@@ -210,27 +210,21 @@ type Proxy struct {
 	// of configuration.
 	XdsResourceGenerator XdsResourceGenerator
 
-	// Active contains the list of watched resources for the proxy, keyed by the DiscoveryRequest type.
-	// It is nil if the Proxy uses the default generator
+	// Active contains the list of watched resources for the proxy, keyed by the canonical DiscoveryRequest type.
+	// Note that the key may not be equal to the proper TypeUrl. For example, Envoy types like Cluster will share a single
+	// key for multiple versions.
 	Active map[string]*WatchedResource
```

Comment:
We should probably define a proper canonical discovery request type enum like `DiscoveryRequestType` /`DiscoveryServiceType` and then a have a mapping  of TypeUrl -> DiscoveryRequestType so that most of the GetShortName etc can be avoided?

Reply:
I don't think we want an enum since we can handle arbitrary types now with the istio types we send. So we just need a translation from string to string which is what this does (that is why the default is return the original). That was the intent anyways